logger/options.go

```diff
@@ -40,43 +41,71 @@ type Options struct {
 	Timestamp     time.Time
 }
 
-func (e InvalidOptionMessage) String() string {
-	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
+type fieldIssue struct {
+	field string
+	prob  string
 }
 
-func validateOptionLength(option string, value string, problems *[]string) {
-	if len(value) > 32 {
-		*problems = append(*problems, InvalidOptionMessage{option, "length must be less than 32"}.String())
+type optionsError struct {
+	issues []fieldIssue
+}
+
+func (e *optionsError) Error() string {
+	var str strings.Builder
+	str.WriteString("One or more invalid options:\n")
+	for i := 0; i < len(e.issues); i++ {
+		str.WriteString(fmt.Sprintf("%s: %s\n", e.issues[i].field, e.issues[i].prob))
 	}
+	return str.String()
 }
 
-func (options *Options) validate() error {
-	var problems []string
-	reMacAddress := regexp.MustCompile(`^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])`)
+func validateOptionLength(option string, value string) *fieldIssue {
+	if len(value) > maxOptionLength {
+		return &fieldIssue{field: option, prob: "length must be less than 80"}
+	}
+	return nil
+}
+
+func (options *Options) validate(flag bool) error {
 	reHostname := regexp.MustCompile(`(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])`)
```

Comment:
The jira ticket talks about the regex for hostname being wrong. Is this the relevant one?

Reply:
am using the same hostname_check that we have in logger_node: HOSTNAME_CHECK: /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$/

https://github.com/logdna/logger-node/blob/master/lib/constants.js#L14

Reply:
@mdeltito is this what you were referring to in the Jira task?

Reply:
it's not quite the same - it's missing the leading and trailing `^`and `$` which makes it match _any string_ as long as a single valid character exists.

Reply:
current version: https://play.golang.org/p/yn5gn7YPZHp

Reply:
fixed!